src/jitlog/reader.lua

```diff
@@ -65,6 +65,37 @@ function base_actions:alltraceflush(msg)
   return flush
 end
 
+local gcstates  = {
+  [0] = "pause", 
+  "propagate", 
+  "atomic",
+  "sweepstring", 
+  "sweep", 
+  "finalize",
+}
+
+function base_actions:gcstate(msg)
+  local newstate = msg:get_state()
+  local prevstate = msg:get_prevstate()
+  local oldstate = self.gcstateid
+  self.gcstateid = newstate
+  self.gcstate = gcstates[newstate]
+  self.gcscount = self.gcscount + 1
+  
+  if oldstate ~= newstate then
+    -- A new gc cycle has only started once were past 'pause'
```

Comment:
"were" -> "we're". I'm not totally sure what "pause" refers to -- is that an internal GC state or ...?

Reply:
The pause is the GC state where its not running/its finished all the phases it goes pause, propagate,  atomic, sweepstring,  sweep, finalize.

Reply:
Is it really "paused" or just "not running"?

Reply:
Well its not doing anything until the GC heap grows past a certain size

Reply:
I think that's "not running" then :)

Reply:
comment updated slightly